### PR TITLE
Table_to_hdu unit test

### PR DESCRIPTION
**Pull request by evertrol**
Amendments for #4778

### CODE REVIEW

File: astropy/io/fits/tests/test_convenience.py [hunk 1]
```diff
@@ -23,3 +26,13 @@ def test_resource_warning(self):
         with catch_warnings() as w:
             header = fits.getheader(self.data('test0.fits'))
             assert len(w) == 0
+
+    def test_table_to_hdu(self, tmpdir):
+        data = np.array(
+            list(zip([1, 2, 3], ['a', 'b', 'c'], [2.3, 4.5, 6.7])),
+            dtype=[(str('a'), int), (str('b'), str('U1')), (str('c'), float)])
+        table = Table(data)
```
Comment (taldcroft) [inline astropy/io/fits/tests/test_convenience.py]: The "astropy" way of doing this (IMHO) is cleaner and more readable:
```
table = Table([[1, 2, 3], ['a', 'b', 'c'], [2.3, 4.5, 6.7]], 
              names=['a', 'b', 'c'], dtype=['i', 'U1', 'f'])
```
And note that `Table` allows unicode for `names` and `dtype` so no `str()` casting needed.

File: astropy/io/fits/convenience.py [hunk 2]
```diff
@@ -435,7 +435,7 @@ def table_to_hdu(table):
     if table.has_mixin_columns:
         mixin_names = [name for name, col in table.columns.items()
                        if not isinstance(col, table.ColumnClass)]
-        raise ValueError('cannot write table with mixin column(s) {0} to FITS'
+        raise ValueError('cannot create table with mixin column(s) {0} to FITS'
```
Comment (taldcroft) [inline astropy/io/fits/convenience.py]: Just changing `write` to `create` confusing and not quite grammatical.

**Comment by taldcroft** [comment]
The changed exception message needs to be matched by changes in the tests.  It's important to always run tests locally and make sure they pass before pushing up to GitHub.

**Comment by evertrol** [comment]
@taldcroft The changed exception message actually causes a problem where the assertion tries to catch all variants of FITS, HDF5 and VoTable output errors. The issue is largely semantic, in whether it's about writing a file or simply creating a single HDU when this exception is raised.
I wonder if it's worth putting an if-else for that assertion for FITS files, or whether to change the exception messages back to what they where (most of the times, when one creates a single HDU, it will be written to disk somewhat later anyway).
Alternatively, it could be argued that the issue is never so much about writing to file, as it is about creating the format (in memory) before writing it. Which would essentially mean changing the other exception messages to 'create' (and hope no-one relies on catching the precise error message, which would break things).

**Comment by taldcroft** [comment]
@evertrol - ah, I forgot that this same message appears from several different bits of code.  With that, I would say the best is just to revert to what it was.  Sorry for the detour.

**Comment by pllim** [comment]
Is it too late to tag 1.2? If so, please remove it.

**Comment by taldcroft** [comment]
It's OK to tag as 1.2, and the coordination committee will decide whether it gets accepted.

**Comment by astrofrog** [comment]
Thanks @evertrol!

**Pull request merged**